### ISSUE TITLE
Expose Cloud Provider Region in Asset View

### ISSUE THREAD

**Issue by srpomeroy** [kubecost/cost-analyzer-helm-chart#1656]
**What problem are you trying to solve?**
Identify what region a resource, specifically a node, is running in. 

**Describe the solution you'd like**
It would be helpful to display regions with cloud and instance/vm size in the asset view.

The nodes have a label that provides this detail which Kubecost is already capturing. 

```json
{
    "type": "Node",
    "properties": {...},
    "labels": {
        "node": "ip-192-168-1-62.us-west-2.compute.internal",
        "label_topology_kubernetes_io_region": "us-west-2",
        "label_topology_kubernetes_io_zone": "us-west-2b",
        "label_node_kubernetes_io_instance_type": "m5.xlarge",
        "label_kubernetes_io_arch": "amd64",
        "label_kubernetes_io_hostname": "ip-192-168-1-62.us-west-2.compute.internal",
        "label_kubernetes_io_os": "linux",
        ...
    },
    "window": {...},
    ...
}
```

Most cloud providers have standardized on `topology.kubernetes.io/region`

Examples: 

* AWS: `topology.kubernetes.io/region=us-west-2`
* Azure: `topology.kubernetes.io/region=eastus`
* GCP:  `topology.kubernetes.io/region=us-central1`

**Describe alternatives you've considered**
Cross-referencing with cloud provider console to determine proper region.

**How would users interact with this feature?**
Add an additional bubble to the line item in the asset view and an additional property in the resource modal.
![image](https://user-images.githubusercontent.com/15660532/186450262-f83b0d74-e4d1-460f-923e-4276ea02cf6c.png)

![image](https://user-images.githubusercontent.com/15660532/186450185-bbabe795-da58-4641-bf17-88c776a4787c.png)

gz#2477<br><br><i>(related to <a href='https://kubecost.zendesk.com/agent/tickets/2477'>Zendesk ticket #2477</a>)</i>

┆Issue is synchronized with this [Jira Task](https://kubecost.atlassian.net/browse/GIB-161) by [Unito](https://www.unito.io)

**Comment by dwbrown2** [comment]
+1, @srpomeroy know if this information would be available for nodes that aren't actively monitored, i.e. would this also be in the CUR?

**Comment by srpomeroy** [comment]
@dwbrown2 Yes, there is a label added by the providers to the nodes `topology.kubernetes.io/region`. I am not sure if this label is added if not using a managed Kubernetes offering. Otherwise, the region information exists in all provider bills.

Examples:

- AWS: `product_region`
- Azure: `resourceLocation` OR `location`
- GCP: `location.location` OR `location.region`

**Comment by srpomeroy** [comment]
I could see value in being able to filter or aggregate by region as well.

**Comment by jessegoodier** [comment]
I have had a couple more customers ask for this. a node label exists in all known providers:
` topology.kubernetes.io/region`

**Comment by nikovacevic** [comment]
The title of this ticket is "Expose Cloud Provider Region in Asset View" but this screenshot is from Allocation, right? At the bottom of drill-down?

**Comment by nikovacevic** [comment]
There is a draft PR out that adds these labels to Allocation APIs: https://github.com/opencost/opencost/pull/1618

Data is just beginning to flow, but here's an example of aggregating by `label:topology_kubernetes_io_region` http://a74e2e639cf144b92bc63baaecad4d08-1675674774.us-east-2.elb.amazonaws.com:9090/allocations?idle=separate&agg=label%3Atopology_kubernetes_io_region

Anything you particularly want to see a proof-of-concept for, while I'm in here making these changes? cc @jessegoodier @srpomeroy

**Comment by nikovacevic** [comment]
cc @nealormsbee just getting this on your radar so that we can close the loop on the UI work after the API work is merged.

**Comment by srpomeroy** [comment]
> The title of this ticket is "Expose Cloud Provider Region in Asset View" but this screenshot is from Allocation, right? At the bottom of drill-down?

Negative. Assets > Aggregate by: Service > Kubernetes > Nodes > Select a node.

> Anything you particularly want to see a proof-of-concept for while I'm in here making these changes?

Does this also mean we will be able to filter by that label in the savings reports that use the allocation API?

**Comment by srpomeroy** [comment]
> Does this also mean we will be able to filter by that label in the savings reports that use the allocation API?

Duh Sean, there's a test environment. See for yourself.
http://a74e2e639cf144b92bc63baaecad4d08-1675674774.us-east-2.elb.amazonaws.com:9090/request-sizing?filters=label%3Atopology_kubernetes_io_region%253Aus-east-2

**Comment by nikovacevic** [comment]
Oh, I see. For the Assets page, that data is already in the payload. Maybe @nealormsbee can give an estimate for how long it would take to pull that data into a little visual element. (i.e. `label_topology_kubernetes_io_zone: "us-east-2c"`)

And ha! Yeah, theoretically the labels should flow through to that view. But curiously that link doesn't apply the filters for me, and the view element that should let you set them isn't working for me. Maybe grounds for a ticket? Or am I just a dummy? (Don't answer that... I already know I'm a dummy.)

**Comment by srpomeroy** [comment]
Perhaps because I was on legacy UI?

**Comment by teevans** [comment]
PR Opened here: https://github.com/kubecost/cost-analyzer-frontend/pull/1541

**Comment by srpomeroy** [comment]
closed by kubecost/cost-analyzer-frontend/#1541

**Comment by srpomeroy** [comment]
closed by kubecost/cost-analyzer-frontend#1541
